x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -140,6 +141,8 @@
 public final class SamlRealm extends Realm implements Releasable {
     private static final Logger logger = LogManager.getLogger(SamlRealm.class);
 
+    private static final AtomicReference<Set<PublicKey>> PREVIOUS_CREDENTIALS_REF = new AtomicReference<>(null);
```

Comment:
Hm, actually I just noticed this. I don't love having this `static` variable here. And since (I think) we can create multiple SAML realms, doesn't this introduce a race condition?

Reply:
Good catch. It should not be static.

Reply:
Fixed